tasks/pin_compiler.rb

```diff
@@ -0,0 +1,32 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+require_relative '../../ruby_task_helper/files/task_helper.rb'
+require 'puppet'
+require 'net/http'
+
+Puppet.initialize_settings
+
+# Turn comma separated agents into an array
+agents = JSON.parse(ARGF.read)['agent_certnames'].split(',')
+uri = URI.parse('https://localhost:4433/classifier-api/v1/groups/')
+
+# Create a Net::HTTP object and set the auth to use our certificates
+http = Net::HTTP.new(uri.host, uri.port)
+http.use_ssl = true
+http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+http.cert = OpenSSL::X509::Certificate.new(File.read(Puppet.settings['hostcert']))
+http.key = OpenSSL::PKey::RSA.new(File.read(Puppet.settings['hostprivkey']))
+http.ca_file = Puppet.settings['localcacert']
+
+# Get the id of the 'PE Master' node group
+uri = URI.parse('https://localhost:4433/classifier-api/v1/groups')
+res = http.request_get(uri.request_uri).read_body
+master_id = JSON.parse(res).find { |group| group['name'] == 'PE Master' }['id']
+
+# use the /pin endpoint to pin our compile masters to PE Master
+uri = URI.parse('https://localhost:4433/classifier-api/v1/groups/%s/pin' % master_id)
+req = Net::HTTP::Post.new(uri)
+req.content_type = 'application/json'
+req.body = { 'nodes' => agents }.to_json
+
+http.request(req)
+print({ 'status' => 'success' }.to_json)
```

Comment:
What happens when the request is not successful?

Reply:
It returns the error from the Ruby library, which is printed out by the calling plan.

Reply:
I guess the question is more if we should catch the exception, pack it into json and fail the task. Just another nit/TODO

Reply:
Honestly I was leaving the error handling for later, we could make a feature branch for that.